Public/Src/Tools/Execution.Analyzer/Analyzers.Core/Xldb/XLGToDBAnalyzer.cs

```diff
@@ -594,6 +665,118 @@ public void IngestAllPips()
             });
         }
 
+        /// <summary>
+        /// Ingest file and directory producer and consumer information to RocksDB
+        /// </summary>
+        private void IngestProducerConsumerInformation()
+        {
+            var parallelOptions = new ParallelOptions();
+            parallelOptions.MaxDegreeOfParallelism = Environment.ProcessorCount;
+
+            Parallel.ForEach(m_fileConsumerMap, parallelOptions, kvp =>
+            {
+                var fileConsumerKey = new FileProducerConsumerKey()
+                {
+                    Type = ProducerConsumerType.Consumer,
+                    FileArtifact = kvp.Key.Path.ToString(PathTable, PathFormat.Windows),
+                    RewriteCount = kvp.Key.RewriteCount
+                };
+
+                var fileConsumerValue = new FileConsumerValue();
+                fileConsumerValue.PipIds.AddRange(kvp.Value);
+
+                WriteToDb(fileConsumerKey.ToByteArray(), fileConsumerValue.ToByteArray(), XldbDataStore.StaticGraphColumnFamilyName);
+            });
+
+            Parallel.ForEach(m_directoryConsumerMap, parallelOptions, kvp =>
+            {
+                var directoryConsumerKey = new DirectoryProducerConsumerKey()
+                {
+                    Type = ProducerConsumerType.Consumer,
+                    DirectoryArtifact = kvp.Key.Path.ToString(PathTable, PathFormat.Windows)
+                };
+
+                var directoryConsumerValue = new DirectoryConsumerValue();
+                directoryConsumerValue.PipIds.AddRange(kvp.Value);
+
+                WriteToDb(directoryConsumerKey.ToByteArray(), directoryConsumerValue.ToByteArray(), XldbDataStore.StaticGraphColumnFamilyName);
+            });
+
+            Parallel.ForEach(CachedGraph.PipGraph.AllFilesAndProducers, parallelOptions, kvp =>
+            {
+                var fileProducerKey = new FileProducerConsumerKey()
+                {
+                    Type = ProducerConsumerType.Producer,
+                    FileArtifact = kvp.Key.Path.ToString(PathTable, PathFormat.Windows),
+                    RewriteCount = kvp.Key.RewriteCount
+                };
+
+                var fileProducerValue = new FileProducerValue()
+                {
+                    PipId = kvp.Value.Value
+                };
+
+                WriteToDb(fileProducerKey.ToByteArray(), fileProducerValue.ToByteArray(), XldbDataStore.StaticGraphColumnFamilyName);
+            });
+
+            Parallel.ForEach(CachedGraph.PipGraph.AllOutputDirectoriesAndProducers, parallelOptions, kvp =>
+            {
+                var directoryProducerKey = new DirectoryProducerConsumerKey()
+                {
+                    Type = ProducerConsumerType.Producer,
+                    DirectoryArtifact = kvp.Key.Path.ToString(PathTable, PathFormat.Windows)

```

Comment:
>Path.ToString(PathTable, PathFormat.Windows) [](start = 48, length = 44)

I'd a helper function that does AP->string conversion. alternatively, I'd create a const for `PathFormat.Windows` and use that